### PR TITLE
refactored primitive obsession when getting file name from path to ex…

### PR DESCRIPTION
**Pull request by lprimak**
…isting java File() class

## Description
*** please merge latest payara master to get accurate diffs ***

### CODE REVIEW

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 1]
```diff
@@ -1661,14 +1652,12 @@ private void deployAll() throws GlassFishException {
                     try {
                         // Convert the URL to a URI for use with the deploy method
                         URI artefactURI = deploymentMapEntry.getValue().toURI();
-
-                        String artefactName = getFileName(artefactURI.getPath());
-                        String name = artefactName.substring(0, artefactName.length() - 4);
+                        String artefactName = removeJavaArchiveExtension(new File(artefactURI).getName());
```
Comment (avpinchuk) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java:1655]: If the artefactURI does not have the file scheme will we not get IllegalArgumentException?

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 2]
```diff
@@ -2084,7 +2080,7 @@ private void getGAVURLs() throws GlassFishException {
             }
 
             for (Map.Entry<String, URL> deploymentMapEntry : deploymentURLsMap.entrySet()) {
-                String artefactName = getFileName(deploymentMapEntry.getValue().getPath());
+                String artefactName = new File(deploymentMapEntry.getValue().toURI()).getName();
```
Comment (avpinchuk) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java:2083]: Same question

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 3]
```diff
@@ -2330,10 +2326,9 @@ private void packageUberJar() {
                 getGAVURLs();
                 for (Map.Entry<String, URL> deploymentMapEntry : deploymentURLsMap.entrySet()) {
                     URL deployment = deploymentMapEntry.getValue();
-                    String name = getFileName(deployment.getPath());
-                    creator.addDeployment(name, deployment);
+                    creator.addDeployment(new File(deployment.toURI()).getName(), deployment);
```
Comment (avpinchuk) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java:2329]: And here

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 4]
```diff
@@ -2330,10 +2326,9 @@ private void packageUberJar() {
                 getGAVURLs();
                 for (Map.Entry<String, URL> deploymentMapEntry : deploymentURLsMap.entrySet()) {
```
Comment (avpinchuk) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java:2327]: May be change to deploymentsURLsMap.values() bause keys is not used?

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java (hunk 1 above)
Comment (lprimak) [reply appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java:1655]: Don’t know. Please test and change as appropriate

**Pull request merged**